src/python/grpcio/grpc/experimental/aio/_interceptor.py

```diff
@@ -133,103 +169,67 @@ class InterceptedUnaryUnaryCall(_base_call.UnaryUnaryCall):
     intercepted call, being at the same time the same call returned to the
     interceptors.
 
-    For most of the methods, like `initial_metadata()` the caller does not need
-    to wait until the interceptors task is finished, once the RPC is done the
-    caller will have the freedom for accessing to the results.
-
-    For the `__await__` method is it is proxied to the intercepted call only when
-    the interceptor task is finished.
+    As a base class for all of the interceptors implements the logic around
+    final status, metadata and cancellation.
     """
 
-    _loop: asyncio.AbstractEventLoop
-    _channel: cygrpc.AioChannel
-    _cancelled_before_rpc: bool
-    _intercepted_call: Optional[_base_call.UnaryUnaryCall]
-    _intercepted_call_created: asyncio.Event
     _interceptors_task: asyncio.Task
     _pending_add_done_callbacks: Sequence[DoneCallbackType]
 
-    # pylint: disable=too-many-arguments
-    def __init__(self, interceptors: Sequence[UnaryUnaryClientInterceptor],
-                 request: RequestType, timeout: Optional[float],
-                 metadata: MetadataType,
-                 credentials: Optional[grpc.CallCredentials],
-                 wait_for_ready: Optional[bool], channel: cygrpc.AioChannel,
-                 method: bytes, request_serializer: SerializingFunction,
-                 response_deserializer: DeserializingFunction,
-                 loop: asyncio.AbstractEventLoop) -> None:
-        self._channel = channel
-        self._loop = loop
-        self._interceptors_task = loop.create_task(
-            self._invoke(interceptors, method, timeout, metadata, credentials,
-                         wait_for_ready, request, request_serializer,
-                         response_deserializer))
+    def __init__(self, interceptors_task: asyncio.Task) -> None:
+        self._interceptors_task = interceptors_task
         self._pending_add_done_callbacks = []
         self._interceptors_task.add_done_callback(
-            self._fire_pending_add_done_callbacks)
+            self._fire_or_add_pending_add_done_callbacks)
```

Comment:
Nit: This should be called `_fire_or_add_pending_done_callbacks`. The method `add_done_callbacks` adds `done_callback`s. That is, these callbacks are called "done callbacks", not "add done callbacks".